src/main/java/org/truffleruby/core/regexp/MatchDataNodes.java

```diff
@@ -259,12 +259,13 @@ protected RubyArray getIndex(RubyMatchData matchData, int index, int length,
                         "getRegexp(matchData) == regexp",
                         "cachedIndex == index" })
         protected Object getIndexSymbolSingleMatch(RubyMatchData matchData, RubySymbol index, NotProvided length,
+                @Cached ConditionProfile singleBackRefProfile,
                 @Cached("index") RubySymbol cachedIndex,
                 @Cached("getRegexp(matchData)") RubyRegexp regexp,
                 @Cached("findNameEntry(regexp, index)") NameEntry name,
                 @Cached("numBackRefs(name)") int backRefs,
                 @Cached("backRefIndex(name)") int backRefIndex) {
-            if (backRefs == 1) {
+            if (singleBackRefProfile.profile(backRefs == 1)) {
```

Comment:
This doesn't need a profile, `backRefs` is `@Cached` and so PE-constant.

Reply:
Good catch. My pattern matching was over-eager.